### PR TITLE
Redundant boolean is redundant

### PR DESCRIPTION
**Pull request by shadow93**
- Check the currentBGM string directly if it's empty. If no BGM plays, its obviously empty.
- Replace StopSound with StopMusic for !ff2music

### CODE REVIEW

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 1]
```diff
@@ -7900,15 +7893,17 @@ public MusicTogglePanelH(Handle:menu, MenuAction:action, client, selection)
...
+			StopMusic(client);
 		}
 		else  //On
 		{
 			//If they already have music enabled don't do anything
 			if(!CheckSoundException(client, SOUNDEXCEPT_MUSIC))
 			{
 				SetClientSoundOptions(client, SOUNDEXCEPT_MUSIC, true);
+			}
+			if(!currentBGM[client][0])
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Newline before this if.

**Comment by 50Wliu** [comment]
I don't think this solves the don't-play-music-after-a-new-round-starts bug:
1. `ff2_stop_music`, `currentBGM` gets cleared
2. Round ends
3. Round starts and `Timer_PlayBGM` is called
4. `currentBGM` is empty, music gets played (not supposed to happen)

**Comment by shadow93** [comment]
What exactly do you mean by "don't-play-music-after-a-new-round-starts bug"?

**Comment by 50Wliu** [comment]
If `ff2_stop_music` is executed music should not be played after a new round starts, only when `ff2_start_music` is.

**Comment by shadow93** [comment]
Ah, that's simple to do

**Comment by shadow93** [comment]
This should be simple to do. Just setting the string value to 'ff2_stop_music' and check if the string is equal to this.

**Comment by shadow93** [comment]
So this 'bug' is caused by this:

```
	CreateTimer(0.2, BossTimer, _, TIMER_REPEAT|TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, CheckAlivePlayers, _, TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, StartRound, _, TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, ClientTimer, _, TIMER_REPEAT|TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(2.0, Timer_PlayBGM, 0, TIMER_FLAG_NO_MAPCHANGE);
```

The client's bool value is ignored because we're passing client index 0!

**Comment by shadow93** [comment]
Finally fixed it.

**Comment by 50Wliu** [comment]
Timer_PlayBGM should happen after PrepareBGM, not the other way around.

**Comment by shadow93** [comment]
PrepareBGM is the contents of what was Timer_PlayBGM to actually play the BGM

**Comment by shadow93** [comment]
Although PrepareBGM can be called directly, it's only an issue for the FF2_StartMusic native, and any function that calls index 0 to play to all.

**Comment by shadow93** [comment]
For that to happen, requires me to do change this:
```
	playing=0;
	for(new client=1; client<=MaxClients; client++)
	{
		if(IsValidClient(client) && !IsBoss(client) && IsPlayerAlive(client))
		{
			playing++;
			CreateTimer(0.15, MakeNotBoss, GetClientUserId(client), TIMER_FLAG_NO_MAPCHANGE);  //TODO:  Is this needed?
		}
	}

	CreateTimer(0.2, BossTimer, _, TIMER_REPEAT|TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, CheckAlivePlayers, _, TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, StartRound, _, TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, ClientTimer, _, TIMER_REPEAT|TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(2.0, Timer_PlayBGM, 0, TIMER_FLAG_NO_MAPCHANGE);
```

to

```
	playing=0;
	for(new client=1; client<=MaxClients; client++)
	{
		if(IsValidClient(client)
		{
			if(!IsBoss(client) && IsPlayerAlive(client))
			{
				playing++;
				CreateTimer(0.15, MakeNotBoss, GetClientUserId(client), TIMER_FLAG_NO_MAPCHANGE);  //TODO:  Is this needed?
			}
			PrepareBGM(client);
		}
	}

	CreateTimer(0.2, BossTimer, _, TIMER_REPEAT|TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, CheckAlivePlayers, _, TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, StartRound, _, TIMER_FLAG_NO_MAPCHANGE);
	CreateTimer(0.2, ClientTimer, _, TIMER_REPEAT|TIMER_FLAG_NO_MAPCHANGE);
```

**Comment by shadow93** [comment]
I need to undo that last commit, the 2 second timer is needed.

**Comment by shadow93** [comment]
I need to undo that last commit

**Comment by 50Wliu** [comment]
> Does the 2 second delay when firing Timer_PlayBGM even serve any purpose at all?

Not that I'm aware of.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 2]
```diff
@@ -2870,16 +2868,110 @@ public Action:StartBossTimer(Handle:timer)
...
+		CreateTimer(0.0, Timer_PlayBGM, 0, TIMER_FLAG_NO_MAPCHANGE);
+	}
+	else
+	{
+		StopMusic(client);
+		CreateTimer(0.0, Timer_PlayBGM, GetClientUserId(client), TIMER_FLAG_NO_MAPCHANGE);
+	}
+}
+
+StopMusic(client=0, bool:disable=false)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Can you rename `disable` to `permanent` instead?

Comment (shadow93) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Will do.

**Comment by WakaFloockaFlame** [comment]
Let me know when all music issues are fixed i tried fixing my self and made things worst. And please send the fixed sp to me thanks

**Comment by shadow93** [comment]
You can try compiling it yourself and see if you get issues. Did an overnight non-stop music test and I've yet to get any bugs from the test server. See how it goes with a live server though.

**Comment by WakaFloockaFlame** [comment]
@shadow93 https://github.com/shadow93/FF2-Official/blob/9f12eba912d0373b8eac1766c2d776e5b3ba8a04/addons/sourcemod/scripting/freak_fortress_2.sp do i compile that i diffed it and it's doesn't add all the changes you've done

**Pull request merged**